dashboard/app/controllers/pusher_controller.rb

```diff
@@ -0,0 +1,25 @@
+# Auth endpoint for prototype pusher features
+require 'pusher'
+
+class PusherController < ApplicationController
+  def auth
+    @pusher = Pusher::Client.new(
+      app_id: CDO.pusher_app_id,
+      key: CDO.pusher_application_key,
+      secret: CDO.pusher_application_secret,
+      use_tls: true
+    )
+
+    if current_user
+      render json: @pusher.authenticate(
+        params[:channel_name],
+        params[:socket_id],
+        {
+          user_id: current_user.id
+        }
+      )
+    else
+      render text: 'Forbidden', status: '403'
+    end
```

Comment:
Note, we're doing a naive implementation here, but since the channel name is given as a parameter we could do proper auth restricting subscriptions to channels for your own sections, etc.